source/developers-guide/customer-streams-extension/index.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: default
+title: Customer - Search and Streams
+github_link: developers-guide/customer-streams-extension/index.md
+indexed: true
+group: Developer Guides
+subgroup: General Resources
+menu_title: Customer - Search & Streams
+menu_order: 500
+---
+
+This document describes how to extend the customer stream and customer search part in shopware. In most cases, a plugin wants to provide more conditions or data for the search. This document describes, by way of a simple example, how this is possible.
+
+## Add own condition
+In order to extend the search with a further condition, the same procedure can be used as for the product search. The customer search is given a criteria object in which the different search conditions are summarized. First, it is necessary to define your own condition:
+
+```
+<?php
+
+namespace SwagCustomerSearchExtension\Bundle\CustomerSearchBundle;
+
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+
+class ActiveCondition implements ConditionInterface
+{
+    /**
+     * @var bool
+     */
+    protected $active;
+
+    /**
+     * @param bool $active
+     */
+    public function __construct($active)
+    {
+        $this->active = $active;
+    }
+
+    public function getName()
+    {
+        return 'ActiveCondition';
+    }
+
+    public function onlyActive()
+    {
+        return $this->active;
+    }
+}
+```
+This condition describes only on an abstract level what is to be searched for. The actual processing of the condition then happens in the corresponding implementation of the search. Currently the customer search in Shopware is only executed in SQL. Therefore, a handler class is written below, which handles this condition in SQL.
```

Comment:
Alternativer Vorschlag: The condition above only describes on an abstract level what is to be searched for. The actual processing of the condition happens in the corresponding implementation of the search. Currently the customer search in Shopware is only executed in SQL. An actual handler class which handles this condition in SQL could look like this: